ray-operator/controllers/ray/common/job.go

```diff
@@ -0,0 +1,119 @@
+package common
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	semver "github.com/Masterminds/semver/v3"
+	"github.com/google/shlex"
+	rayv1alpha1 "github.com/ray-project/kuberay/ray-operator/apis/ray/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+// GetDecodedRuntimeEnv decodes the runtime environment for the Ray job from a base64-encoded string.
+func GetDecodedRuntimeEnv(runtimeEnv string) (string, error) {
+	decodedBytes, err := base64.StdEncoding.DecodeString(runtimeEnv)
+	if err != nil {
+		return "", fmt.Errorf("failed to decode runtimeEnv: %v: %v", runtimeEnv, err)
+	}
+	return string(decodedBytes), nil
+}
+
+// GetBaseRayJobCommand returns the first part of the Ray Job command up to and including the address, e.g. "ray job submit --address http://..."
+func GetBaseRayJobCommand(address string) []string {
+	// add http:// if needed
+	if !strings.HasPrefix(address, "http://") {
+		address = "http://" + address
+	}
+	return []string{"ray", "job", "submit", "--address", address}
+}
+
+// GetMetadataJson returns the JSON string of the metadata for the Ray job.
+func GetMetadataJson(metadata map[string]string, rayVersion string) (string, error) {
+	// Check that the Ray version is at least 2.6.0.
+	// If it is, we can use the --metadata-json flag.
+	// Otherwise, we need to raise an error.
+	constraint, _ := semver.NewConstraint(">= 2.6.0")
```

Comment:
Why are we restricted to using `Ray >= 2.6.0`?

Reply:
The `metadata-json` CLI argument was only introduced in Ray 2.6.0.  Prior to Ray 2.6.0, `metadata` could only be passed via the Ray Jobs REST API or Python SDK.  So this could be a breaking change for KubeRay users who use `metadata` and upgrade kuberay to v0.6.0 but don't upgrade Ray to 2.6.0.  I think it's okay because RayJob is still alpha and `metadata` is not a commonly used field. I don't see an easy workaround unfortunately.